fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -79,13 +106,76 @@ class WebExtensionPromptFeature(
 
     private fun handlePermissionRequest(
         addon: Addon,
-        promptRequest: WebExtensionPromptRequest.Permissions,
+        promptRequest: WebExtensionPromptRequest.AfterInstallation.Permissions,
     ) {
-        if (hasExistingPermissionDialogFragment()) {
-            return
+        if (hasExistingPermissionDialogFragment()) return
+        showPermissionDialog(
+            addon,
+            promptRequest,
+        )
+    }
+
+    @VisibleForTesting
+    internal fun handleInstallationFailedRequest(
+        exception: WebExtensionInstallException,
+    ) {
+        val addonName = exception.extensionName ?: ""
+        var title = context.getString(R.string.mozac_feature_addons_failed_to_install, "")
+        val message = when (exception) {
+            is WebExtensionInstallException.Blocklisted -> {
+                context.getString(R.string.mozac_feature_addons_blocklisted, addonName)
+            }
+
+            is WebExtensionInstallException.Unknown, is WebExtensionInstallException.UserCancelled -> {
+                // Making sure we don't have a
+                // Title = Failed to install
+                // Message = Failed to install $addonName
+                title = ""
+                context.getString(R.string.mozac_feature_addons_failed_to_install, addonName)
+            }
+
+            is WebExtensionInstallException.NetworkFailure -> {
+                context.getString(R.string.mozac_feature_addons_failed_to_install_network_error)
+            }
+
+            is WebExtensionInstallException.CorruptedFile -> {
+                context.getString(R.string.mozac_feature_addons_failed_to_install_corrupt_error)
+            }
+
+            is WebExtensionInstallException.IncorrectHash -> {
+                context.getString(R.string.mozac_feature_addons_failed_to_install_hash_error)
+            }
+
+            is WebExtensionInstallException.FileAccess -> {
+                context.getString(
+                    R.string.mozac_feature_addons_failed_to_install_file_access_error,
+                    addonName,
+                )
+            }
+
+            is WebExtensionInstallException.UnSigned -> {
+                context.getString(
+                    R.string.mozac_feature_addons_failed_to_install_un_signed_error,
+                    addonName,
+                )
+            }
+
+            is WebExtensionInstallException.Incompatible -> {
+                val appName = context.getString(R.string.app_name)
+                val version = context.appVersionName
```

Comment:
I think that issue is because you are using a debug variant, same problem can be seen when going to the about page
<img width="393" alt="image" src="https://github.com/mozilla-mobile/firefox-android/assets/773158/5b818dd0-03fe-40df-8699-25804aca68c3">

When building locally a nightly version I get `119.0a1` for `context.appVersionName` :)